boefjes/boefjes/clients/scheduler_client.py

```diff
@@ -106,4 +92,92 @@ def get_task(self, task_id: uuid.UUID) -> Task:
+        octopoes_api_connector = get_octopoes_api_connector(boefje_meta.organization)
+        input_ooi = boefje_meta.input_ooi
+        boefje_meta.arguments = {"oci_image": plugin.oci_image, "oci_arguments": plugin.oci_arguments}
+        boefje_meta.runnable_hash = plugin.runnable_hash
+
+        if input_ooi:
+            reference = Reference.from_str(input_ooi)
+            try:
+                ooi = octopoes_api_connector.get(reference, valid_time=datetime.datetime.now(datetime.timezone.utc))
+                boefje_meta.arguments["input"] = ooi.serialize()
+            except ObjectNotFoundException:
+                logger.info(
+                    "Can't run boefje because OOI does not exist anymore [reference=%s]",
+                    reference,
+                    boefje_id=boefje_meta.boefje.id,
+                    ooi=boefje_meta.input_ooi,
+                    task_id=boefje_meta.id,
+                )
+                raise
+
+        boefje_meta.environment = get_environment_settings(boefje_meta, plugin.boefje_schema)
+
+        return boefje_meta
+
+
+@cache
+def boefje_env_variables() -> dict:
+    """
+    Return all environment variables that start with BOEFJE_. The returned
+    keys have the BOEFJE_ prefix removed.
+    """
+
+    boefje_variables = {}
+    for key, value in os.environ.items():
+        if key.startswith("BOEFJE_"):
+            boefje_variables[key.removeprefix("BOEFJE_")] = value
+
+    return boefje_variables
+
+
+def get_system_env_settings_for_boefje(allowed_keys: list[str]) -> dict:
+    return {key: value for key, value in boefje_env_variables().items() if key in allowed_keys}
+
+
+def get_environment_settings(boefje_meta: BoefjeMeta, schema: dict | None = None) -> dict[str, str]:
+    try:
+        katalogus_api = str(settings.katalogus_api).rstrip("/")
+        response = httpx.get(
+            f"{katalogus_api}/v1/organisations/{boefje_meta.organization}/{boefje_meta.boefje.id}/settings", timeout=30
+        )
+        response.raise_for_status()
+    except HTTPError:
+        logger.exception("Error getting environment settings")
+        raise
+
+    allowed_keys = schema.get("properties", []) if schema else []
+    new_env = get_system_env_settings_for_boefje(allowed_keys)
+
+    settings_from_katalogus = response.json()
+
+    for key, value in settings_from_katalogus.items():
+        if key in allowed_keys:
+            new_env[key] = value
+
+    # The schema, besides dictating that a boefje cannot run if it is not matched, also provides an extra safeguard:
+    # it is possible to inject code if arguments are passed that "escape" the call to a tool. Hence, we should enforce
+    # the schema somewhere and make the schema as strict as possible.
+    if schema is not None:
+        try:
+            validate(instance=new_env, schema=schema)
+        except ValidationError as e:
+            raise SettingsNotConformingToSchema(boefje_meta.boefje.id, e.message) from e
```

Comment:
If we are removing local env as valid injectable vars, (and make sure all variables are stored in the KATalogus) we can remove this check.
We want to do this to make sure we can deduplicate jobs based on these vars, which is only possibly if all workers execute the same vars, and we know all the vars over all orga's to make sure we *can* deduplicate.
This would also allow us to properly check boefjes requirements and schema validity before enabling them.